docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceApi.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+namespace Drupal\foia_webform;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\file\Entity\File;
+use Drupal\node\NodeInterface;
+use Drupal\webform\WebformInterface;
+use Drupal\webform\WebformSubmissionInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class FoiaSubmissionServiceApi.
+ */
+class FoiaSubmissionServiceApi implements FoiaSubmissionServiceInterface {
+
+  /**
+   * The Agency Lookup service.
+   *
+   * @var \Drupal\foia_webform\AgencyLookupService
+   */
+  protected $agencyLookUpService;
+
+  /**
+   * An HTTP client to post FOIA requests to an Agency Component API endpoint.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * FoiaSubmissionServiceApi constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $httpClient
+   *   The HTTP Guzzle client.
+   * @param \Drupal\foia_webform\AgencyLookupServiceInterface $agencyLookupService
+   *   The Agency Lookup service.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   */
+  public function __construct(ClientInterface $httpClient, AgencyLookupServiceInterface $agencyLookupService, LoggerInterface $logger) {
+    $this->httpClient = $httpClient;
+    $this->agencyLookUpService = $agencyLookupService;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function sendSubmissionToComponent(WebformSubmissionInterface $webformSubmission, WebformInterface $webform, NodeInterface $agencyComponent) {
+    $apiUrl = $agencyComponent->get('field_submission_api')->value;
+
+    if (!UrlHelper::isValid($apiUrl, TRUE)) {
+      $apiUrl = FALSE;
```

Comment:
The `sendSubmissionToComponent` method should return true if the submission was sent, and false if it was not. As such, rather than setting this to false and then performing the `if ($apiUrl)` check below, we should just be returning false here.